libs/net/core/Http/OpenIdConnectRequestClient.cs

```diff
@@ -91,25 +91,27 @@ public class OpenIdConnectRequestClient : HttpRequestClient, IOpenIdConnectReque
         public async Task<string> RequestAccessToken()
         {
             HttpResponseMessage response;
-            if (_accessToken == null || String.IsNullOrWhiteSpace(_accessToken.AccessToken) || (!String.IsNullOrWhiteSpace(_accessToken.RefreshToken) && _tokenHandler.ReadJwtToken(_accessToken.RefreshToken).ValidTo <= DateTime.UtcNow))
+            var now = DateTime.UtcNow;
```

Comment:
Asking for the current datetime multiple times is not performant, and will result in different values.